Solutions/NCSC-NL NDN Cyber Threat Intelligence Sharing/Playbooks/NCSCNLShareSTIXBundle/azuredeploy.json

```diff
@@ -0,0 +1,2798 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "title": "NCSCNLShareSTIXBundle",
+        "description": "This playbook gets triggered every hour and perform the following actions:\n 1. Get all the threat intelligence indicators from Microsoft Sentinel Workspace with given tag.\n 2. Filter all the indicators whose export in not completed.\n 3. Share the STIX Bundle, that includes the 'Indicator, corresponding Identities, Markings and optional Sighting' with the to provided TAXII server.",
+        "prerequisites": [
+            "1. Have the provided TAXII Server Url, Collection ID, your Organization ID, STIX Identity and API Key available before the deployment of the Playbook",
+            "2. Tag the indicators that need to be exported, by default this playbook exports the indicators with tag 'NDN Share', this can be changes during the deployment of playbook. Details on how to tag can be found [here](https://learn.microsoft.com/azure/sentinel/understand-threat-intelligence#view-and-manage-your-threat-indicators)",
+            "3. Tagged indicators that are shared successfully get the tag 'NDN Sharing Complete', by default include a sighting object."
+        ],
+    "postDeployment": [
+        "This playbook needs contributor role on Log Analytics, to read and update threat indicator tags. 1. Go to Log Analytics Workspace resource --> 2. Select Access control (IAM) tab -->3. Add role assignments --> 4. Select Contributor role --> 5. In the Members tab choose 'Assign access to' Managed Identity --> 6. Click on 'Select members' --> 7. Provide correct Subscription and Managed Identity  --> 8. Provide the playbook name in 'Search by name' textbox --> 9. Select the correct identity and click on Select --> 10. Click on 'Review + assign' "
+    ],
+    "prerequisitesDeployTemplateFile": "",
+    "lastUpdateTime": "2025-05-09T12:13:00Z",
+"entities": [],
+"tags": [
+    "Threat Intelligence"
+],
+"support": {
+"tier": "community",
+"armtemplate": "Generated from https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Playbook-ARM-Template-Generator"
+},
+"author": {
+    "name": "NCSC-NL, Microsoft"
+}
+},
+"parameters": {
+"PlaybookName": {
+"defaultValue": "NCSCNLShareSTIXBundle",
+"type": "string"
+},
+"CollectionID": {
+"type": "String",
+"metadata": {
+    "description": "Enter value for CollectionID"
+}
+},
+"Default TLP Label": {
+"type": "String",
+"defaultValue": "TLP:CLEAR",
+"allowedValues": [
+    "TLP:RED",
+    "TLP:AMBER+STRICT",
+    "TLP:AMBER",
+    "TLP:GREEN",
+    "TLP:CLEAR"
+],
+"metadata": {
+    "description": "Enter value for Default TLP Label"
+}
+},
+"OrgnizationUUID": {
+"type": "String",
+"metadata": {
+    "description": "Enter value for OrgnizationUUID"
+}
+},
+"TAXIIServerPassword": {
```

Comment:
TAXIIServerPassword is an optional parameter, according to readme. It would be better if you set "defaultValue" parameter. Like "defaultValue": "<only provide if not using API key>".  This provides instruction to user and sets a value to parameter. Otherwise, it becomes a mandatory parameter.